Skaled full build broken

Looks like patch related changes may have broken the full build of skaled

cmake --build build

 82%] Building CXX object skale-vm/CMakeFiles/skale-vm.dir/main.cpp.o
/d/skaled/skale-vm/main.cpp: In function ‘int main(int, char**)’:
/d/skaled/skale-vm/main.cpp:292:82: error: no matching function for call to ‘dev::eth::EnvInfo::EnvInfo(dev::eth::BlockHeader&, LastBlockHashes&, int, const uint64_t&)’
  292 |         blockHeader, lastBlockHashes, 0 /* gasUsed */, se->chainParams().chainID );
      |                                                                                  ^
In file included from /d/skaled/libevm/VMFace.h:19,
                 from /d/skaled/libethereum/Executive.h:27,
                 from /d/skaled/libskale/State.h:36,
                 from /d/skaled/libethereum/Block.h:37,
                 from /d/skaled/skale-vm/main.cpp:29:
/d/skaled/libevm/ExtVMFace.h:143:5: note: candidate: ‘dev::eth::EnvInfo::EnvInfo(const dev::eth::BlockHeader&, const dev::eth::LastBlockHashesFace&, time_t, const u256&, const u256&, const u256&)’
  143 |     EnvInfo( BlockHeader const& _working, LastBlockHashesFace const& _lh,
      |     ^~~~~~~
/d/skaled/libevm/ExtVMFace.h:143:5: note:   candidate expects 6 arguments, 4 provided
/d/skaled/libevm/ExtVMFace.h:134:5: note: candidate: ‘dev::eth::EnvInfo::EnvInfo(const dev::eth::BlockHeader&, const dev::eth::LastBlockHashesFace&, time_t, const u256&, const u256&)’
  134 |     EnvInfo( BlockHeader const& _working, LastBlockHashesFace const& _lh,
      |     ^~~~~~~
/d/skaled/libevm/ExtVMFace.h:134:5: note:   candidate expects 5 arguments, 4 provided
/d/skaled/libevm/ExtVMFace.h:132:7: note: candidate: ‘dev::eth::EnvInfo::EnvInfo(const dev::eth::EnvInfo&)’
  132 | class EnvInfo {
      |       ^~~~~~~
/d/skaled/libevm/ExtVMFace.h:132:7: note:   candidate expects 1 argument, 4 provided
/d/skaled/libevm/ExtVMFace.h:132:7: note: candidate: ‘dev::eth::EnvInfo::EnvInfo(dev::eth::EnvInfo&&)’
/d/skaled/libevm/ExtVMFace.h:132:7: note:   candidate expects 1 argument, 4 provided
/d/skaled/skale-vm/main.cpp:299:70: error: no matching function for call to ‘dev::eth::SealEngineFace::evmSchedule(int64_t)’
  299 |         auto const latestVersion = se->evmSchedule( envInfo.number() ).accountVersion;
      |                                                                      ^
In file included from /d/skaled/libethashseal/Ethash.h:26,
                 from /d/skaled/skale-vm/main.cpp:26:
/d/skaled/libethcore/SealEngine.h:97:25: note: candidate: ‘virtual dev::eth::EVMSchedule dev::eth::SealEngineFace::evmSchedule(time_t, const u256&) const’
   97 |     virtual EVMSchedule evmSchedule(
      |                         ^~~~~~~~~~~
/d/skaled/libethcore/SealEngine.h:97:25: note:   candidate expects 2 arguments, 1 provided
/d/skaled/skale-vm/main.cpp:313:49: error: no matching function for call to ‘dev::eth::Executive::Executive(skale::State&, const dev::eth::EnvInfo&, dev::eth::SealEngineFace&, int)’
  313 |     Executive executive( state, envInfo, *se, 0 );
      |                                                 ^
In file included from /d/skaled/libskale/State.h:36,
                 from /d/skaled/libethereum/Block.h:37,
                 from /d/skaled/skale-vm/main.cpp:29:
/d/skaled/libethereum/Executive.h:133:5: note: candidate: ‘dev::eth::Executive::Executive(skale::State&, const dev::eth::Block&, unsigned int, const dev::eth::BlockChain&, const u256&, unsigned int, bool)’
  133 |     Executive( skale::State& io_s, Block const& _block, unsigned _txIndex, BlockChain const& _bc,
      |     ^~~~~~~~~
/d/skaled/libethereum/Executive.h:133:5: note:   candidate expects 7 arguments, 4 provided
/d/skaled/libethereum/Executive.h:125:5: note: candidate: ‘dev::eth::Executive::Executive(dev::eth::Block&, const dev::eth::LastBlockHashesFace&, const u256&, unsigned int, bool)’
  125 |     Executive( Block& _s, LastBlockHashesFace const& _lh, const u256& _gasPrice,
      |     ^~~~~~~~~
/d/skaled/libethereum/Executive.h:125:23: note:   no known conversion for argument 1 from ‘skale::State’ to ‘dev::eth::Block&’
  125 |     Executive( Block& _s, LastBlockHashesFace const& _lh, const u256& _gasPrice,
      |                ~~~~~~~^~
/d/skaled/libethereum/Executive.h:118:5: note: candidate: ‘dev::eth::Executive::Executive(dev::eth::Block&, const dev::eth::BlockChain&, const u256&, unsigned int, bool)’
  118 |     Executive( Block& _s, BlockChain const& _bc, const u256& _gasPrice, unsigned _level = 0,
      |     ^~~~~~~~~
/d/skaled/libethereum/Executive.h:118:23: note:   no known conversion for argument 1 from ‘skale::State’ to ‘dev::eth::Block&’
  118 |     Executive( Block& _s, BlockChain const& _bc, const u256& _gasPrice, unsigned _level = 0,
      |                ~~~~~~~^~
/d/skaled/libethereum/Executive.h:105:5: note: candidate: ‘dev::eth::Executive::Executive(skale::State&, const dev::eth::EnvInfo&, const dev::eth::ChainOperationParams&, const u256&, unsigned int, bool)’
  105 |     Executive( skale::State& _s, EnvInfo const& _envInfo, ChainOperationParams const& _chainParams,
      |     ^~~~~~~~~
/d/skaled/libethereum/Executive.h:105:5: note:   candidate expects 6 arguments, 4 provided
/d/skaled/skale-vm/main.cpp:350:70: error: no matching function for call to ‘dev::eth::SealEngineFace::evmSchedule(int64_t)’
  350 |              << t.baseGasRequired( se->evmSchedule( envInfo.number() ) ) << " for transaction, -"
      |                                                                      ^
In file included from /d/skaled/libethashseal/Ethash.h:26,
                 from /d/skaled/skale-vm/main.cpp:26:
/d/skaled/libethcore/SealEngine.h:97:25: note: candidate: ‘virtual dev::eth::EVMSchedule dev::eth::SealEngineFace::evmSchedule(time_t, const u256&) const’
   97 |     virtual EVMSchedule evmSchedule(
      |                         ^~~~~~~~~~~
/d/skaled/libethcore/SealEngine.h:97:25: note:   candidate expects 2 arguments, 1 provided
gmake[2]: *** [skale-vm/CMakeFiles/skale-vm.dir/build.make:76: skale-vm/CMakeFiles/skale-vm.dir/main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1251: skale-vm/CMakeFiles/skale-vm.dir/all] Error

--- a/cmake/EthUtils.cmake
+++ b/cmake/EthUtils.cmake
@@ -57,7 +57,7 @@ macro(eth_add_test NAME)
 	add_custom_target("test.${NAME}"
 		DEPENDS testeth
 		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-		COMMAND ${CMAKE_COMMAND} -DETH_TEST_NAME="${NAME}" -DCTEST_COMMAND="${CTEST_COMMAND}" -P "${ETH_SCRIPTS_DIR}/runtest.cmake"
+                COMMAND ${CMAKE_COMMAND} -DETH_TEST_NAME="${NAME}" -DCTEST_COMMAND="${CMAKE_CTEST_COMMAND}" -P "${ETH_SCRIPTS_DIR}/runtest.cmake"
 	)
 
 endmacro()

--- a/skale-vm/main.cpp
+++ b/skale-vm/main.cpp
@@ -292,6 +292,8 @@ int main( int argc, char** argv ) {
         0 /* gasUsed */, chainParams.chainID );
     EVMSchedule evmSchedule = chainParams.makeEvmSchedule( 0, envInfo.number() );
 
+    state = state.createStateModifyCopy();
+
     Transaction t;
     Address contractDestination( "1122334455667788991011121314151617181920" );
     if ( !code.empty() ) {
@@ -387,5 +389,8 @@ int main( int argc, char** argv ) {
              << '\n';
         cout << "exec time: " << fixed << setprecision( 6 ) << execTime << '\n';
     }
+
+    state.releaseWriteLock();
+
     return 0;
 }

--- a/skale-vm/main.cpp
+++ b/skale-vm/main.cpp
@@ -286,17 +286,18 @@ int main( int argc, char** argv ) {
         }  // Ignore decoding errors.
     }
 
-    unique_ptr< SealEngineFace > se( ChainParams( genesisInfo( networkName ) ).createSealEngine() );
+    ChainParams chainParams( genesisInfo( networkName ) );
     LastBlockHashes lastBlockHashes;
-    EnvInfo const envInfo(
-        blockHeader, lastBlockHashes, 0 /* gasUsed */, se->chainParams().chainID );
+    EnvInfo const envInfo( blockHeader, lastBlockHashes, 0 /*_committedBlockTimestamp*/,
+        0 /* gasUsed */, chainParams.chainID );
+    EVMSchedule evmSchedule = chainParams.makeEvmSchedule( 0, envInfo.number() );
 
     Transaction t;
     Address contractDestination( "1122334455667788991011121314151617181920" );
     if ( !code.empty() ) {
         // Deploy the code on some fake account to be called later.
         Account account( 0, 0 );
-        auto const latestVersion = se->evmSchedule( envInfo.number() ).accountVersion;
+        auto const latestVersion = evmSchedule.accountVersion;
         account.setCode( bytes{ code }, latestVersion );
         std::unordered_map< Address, Account > map;
         map[contractDestination] = account;
@@ -307,10 +308,12 @@ int main( int argc, char** argv ) {
         // data.
         t = Transaction( value, gasPrice, gas, data, 0 );
 
+    t.ignoreExternalGas();  // for tests
+
     state.addBalance( sender, value );
 
-    // HACK 0 here is for gasPrice
-    Executive executive( state, envInfo, *se, 0 );
+    // HACK 1st 0 here is for gasPrice
+    Executive executive( state, envInfo, chainParams, 0, 0 );
     ExecutionResult res;
     executive.setResultRecipient( res );
     t.forceSender( sender );
@@ -346,9 +349,8 @@ int main( int argc, char** argv ) {
     bytes output = std::move( res.output );
 
     if ( mode == Mode::Statistics ) {
-        cout << "Gas used: " << res.gasUsed << " (+"
-             << t.baseGasRequired( se->evmSchedule( envInfo.number() ) ) << " for transaction, -"
-             << res.gasRefunded << " refunded)\n";
+        cout << "Gas used: " << res.gasUsed << " (+" << t.baseGasRequired( evmSchedule )
+             << " for transaction, -" << res.gasRefunded << " refunded)\n";
         cout << "Output: " << toHex( output ) << "\n";
         LogEntries logs = executive.logs();
         cout << logs.size() << " logs" << ( logs.empty() ? "." : ":" ) << "\n";

